CONTRIBUTING.md

```diff
@@ -0,0 +1,619 @@
+
+# TinyMCE documentation tech stack
+
+The documentation is built with [Antora](https://antora.org/) and text files are written in [ascii doc](https://asciidoc.org/). You will find the [repo here](https://github.com/tinymce/tinymce-docs).
+
+## Highlighting
+
+Code language syntax and associated elements are highlighted using an open and closing backtick. For example, this `object` is actually enclosed with backticks like this: `object`.
+
+Highlight words relating specifically to executable parts of software languages. Keys and their values when pertaining to objects and methods (particularly the case with TinyMCE) should have backticks. We cannot cover every use case here and contributors need to use their best judgment. What is more important is your ability to articulate complex ideas.
+
+## Code snippets
+
+Code snippets should always be complete, executable code blocks starting with the `tinymce` object. We write code snippets by enclosing the example code within three opening and closing backticks. The first set of backticks is immediately followed by the code’s language, which helps correct syntax highlighting: typically, html, css, js (and sometimes php).
+
+Include any key values pairs and method options that are required to enable a user to test the snippet in a local HTML file. You do not need to include the `html` surrounding the `tinymce` object that would be required to make the snippet work.
+
+For example, this is good:
+```js
+    tinymce.init({
+      selector: 'textarea', // change this value according to your HTML
+      plugins: 'codesample',
+      toolbar: 'codesample',
+    });
+```
+
+This is not:
+```js
+    tinymce.init({
+      ...
+      plugins: 'codesample',
+      toolbar: 'codesample'
+      ...
+    });
+```
+
+## URLs and links
+
+Resources to be linked take the form of a URL following by the linked text between two brackets. Linking an external resource looks like this:
+
+```
+    hello, http://www.example.com[text to link] to the URL example.com.
+```
+
+## Live demos
+
+New live demos can be added to [modules/ROOT/examples/live-demos directory](modules/ROOT/examples/live-demos). It then can be referenced in your doc with the following code:
+
+```
+    liveDemo::{sub-directory-name}[]
+```
+
+
+## Adding metadata to a page
+
+Every page should have a meta section starting at line 1 containing information that is both informative and system critical. If you are creating a page or editing one that currently has inadequate meta information, please read the comments next to the default meta fields below.
+
+```
+    :navtitle:          // Descriptive title for the navigation bar, if omitted nav defaults to title field.
+    :description_short: // Short description for the grid layouts.
```

Comment:
```suggestion
```
This isn't a valid property in AsciiDoc and was a custom one used by the old system.